packages/lib/src/vis/tiles/utils.ts

```diff
@@ -50,32 +47,51 @@ export function sortTilesByDistanceTo(
   });
 }
 
-export function getScaledVisibleDomains(
+const NDC_BOX = new Box3(new Vector3(-1, -1, -1), new Vector3(1, 1, 1));
+
+function getObject3DVisibleBox(
+  camera: Camera,
+  context: AxisSystemContextValue,
+  Object3DRef: RefObject<Object3D>
+): Box3 | undefined {
+  const object3D = Object3DRef.current;
+  if (!object3D) {
+    return undefined;
+  }
+
+  // Convert view box: Normalized Device Coordinates -> camera -> world -> local
+  const matrix = new Matrix4()
+    .multiplyMatrices(object3D.matrixWorld.invert(), camera.matrixWorld)
+    .multiply(camera.projectionMatrixInverse);
+  return NDC_BOX.clone().applyMatrix4(matrix);
+}
+
+export function getScaledVisibleBox(
   camera: Camera,
   context: AxisSystemContextValue,
-  size: Size
-): {
-  xVisibleDomain: Domain;
-  yVisibleDomain: Domain;
-} {
-  const { abscissaConfig, ordinateConfig } = context;
-  const { width, height } = size;
-  const { xVisibleDomain, yVisibleDomain } = getVisibleDomains(camera, context);
+  meshSize: Size,
+  arraySize: Size,
+  ref: RefObject<Object3D>
+): Box2 | undefined {
+  const box3d = getObject3DVisibleBox(camera, context, ref);
+  if (!box3d) {
+    return undefined;
+  }
 
-  const xScale = createAxisScale(abscissaConfig.scaleType ?? ScaleType.Linear, {
-    domain: abscissaConfig.visDomain,
-    range: [0, width],
+  const xScale = createAxisScale(ScaleType.Linear, {
+    domain: [-meshSize.width / 2, meshSize.width / 2],
+    range: [0, arraySize.width],
     clamp: true,
   });
 
-  const yScale = createAxisScale(ordinateConfig.scaleType ?? ScaleType.Linear, {
-    domain: ordinateConfig.visDomain,
-    range: [0, height],
+  const yScale = createAxisScale(ScaleType.Linear, {
+    domain: [-meshSize.height / 2, meshSize.height / 2],
+    range: [0, arraySize.height],
     clamp: true,
   });
 
-  return {
-    xVisibleDomain: xVisibleDomain.map(xScale) as Domain,
-    yVisibleDomain: yVisibleDomain.map(yScale) as Domain,
-  };
+  return new Box2().setFromPoints([
```

Comment:
It's more generic because the constructor requires `min`, `max` while `setFromPoints` computes the min/max... but you're right, here that is not needed.

Reply:
I simplified it